form/TimeTextBox.js

```diff
@@ -46,6 +47,18 @@ define([
 		openDropDown: function(/*Function*/ callback){
 			this.inherited(arguments);
 
+                        // Fix #18683
+                        var selectedNode = query(".dijitTimePickerItemSelected", this.dropDown.domNode),
+                            parentNode=this.dropDown.domNode.parentNode;
+                       if (selectedNode[0]) {
+                             // Center the selected node in the client area of the popup.
+                             parentNode.scrollTop=selectedNode[0].offsetTop-parentNode.clientHeight/2;
+                       } else {
+                             //      There is no currently selected value. Position the list so that the median
+                             //      node is visible.
+                             parentNode.scrollTop=(parentNode.scrollHeight-parentNode.clientHeight)/2;
```

Comment:
Likewise, shouldn't this take into account the height of the median node?

Actually, why not start all of this by just saying something like:

```
var nodeToCenter = selectedNode[0] || medianNode;
```

... and then there's no if/else statement.